### PR TITLE
gdtoolkit: init at 3.2.8

### PR DESCRIPTION
**Pull request by critbase**
###### Motivation for this change
To package https://github.com/Scony/godot-gdscript-toolkit, an "independent set of GDScript tools."

Had this branch laying around for a while, forgot to upstream it until now.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
You haven't referenced it from the top-level anywhere (e.g. `all-packages.nix`) so as it is this package is unusable.

File: pkgs/development/tools/gdtoolkit/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  '';
+
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = "https://github.com/Scony/godot-gdscript-toolkit";
+    description =
+      "Independent set of GDScript tools - parser, linter and formatter";
+    license = licenses.mit;
+    platforms = platforms.linux;
```
Comment (risicle) [inline pkgs/development/tools/gdtoolkit/default.nix:28]: Seems to work fine for me on darwin too so you can probably safely change this to `platforms.unix`.

**Comment by critbase** [comment]
Argh, I knew I forgot something. Fixed, thank you. This time I tested to make sure `nix-build -A gdtoolkit` works.
I also added myself to the maintainer-list, because I apparently forgot to.

**Review by risicle — APPROVED** [review]
Builds & runs, macos 10.15 & nixos x86_64

File: pkgs/development/tools/gdtoolkit/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  preConfigure = ''
+    sed -i 's/lark-parser==0.8.0/lark-parser>=0.8.0/' setup.py
+  '';
+
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = "https://github.com/Scony/godot-gdscript-toolkit";
+    description =
+      "Independent set of GDScript tools - parser, linter and formatter";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/gdtoolkit/default.nix:26]:
```suggestion
    description = "Independent set of GDScript tools - parser, linter and formatter";
```

File: pkgs/development/tools/gdtoolkit/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    inherit pname version;
+    sha256 = "3+zr1JcCt8HPwDG0j+ttnzGOhvUmaz+qzbE2wGyT86M=";
+  };
+
+  buildInputs = [ lark-parser docopt pyyaml setuptools ];
+  propagatedBuildInputs = [ lark-parser docopt pyyaml setuptools ];
+
+  preConfigure = ''
+    sed -i 's/lark-parser==0.8.0/lark-parser>=0.8.0/' setup.py
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/tools/gdtoolkit/default.nix:19]:
```suggestion
  postPatch = ''
    substituteInPlace setup.py \
      --replace "lark-parser==0.8.0" "lark-parser>=0.8.0"
  '';
```

File: pkgs/development/tools/gdtoolkit/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  pname = "gdtoolkit";
+  version = "3.2.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "3+zr1JcCt8HPwDG0j+ttnzGOhvUmaz+qzbE2wGyT86M=";
+  };
+
+  buildInputs = [ lark-parser docopt pyyaml setuptools ];
+  propagatedBuildInputs = [ lark-parser docopt pyyaml setuptools ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/gdtoolkit/default.nix:15]:
```suggestion
  propagatedBuildInputs = [ lark-parser docopt pyyaml setuptools ];
```

**Comment by critbase** [comment]
Just found out why this package required lark-parser 0.8.0... just pushed a version that overrides the version of lark-parser used, and made sure to test the binaries on actual `.gd` files.. I should really test my own PRs a lot more thoroughly.
Please review this version.

File: pkgs/development/tools/gdtoolkit/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  version = "3.2.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "3+zr1JcCt8HPwDG0j+ttnzGOhvUmaz+qzbE2wGyT86M=";
+  };
+
+  propagatedBuildInputs = [ lark-parser080 docopt pyyaml setuptools ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/gdtoolkit/default.nix:26]: Please comment why tests are disabled? Can we add a pythonImportsCheck instead?

File: pkgs/development/tools/gdtoolkit/default.nix [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/Scony/godot-gdscript-toolkit";
+    description = "Independent set of GDScript tools - parser, linter and formatter";
+    license = licenses.mit;
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/gdtoolkit/default.nix:32]:
```suggestion
```
set by buildPythonApplication

**Comment by hesiod** [comment]
This PR can be closed, gdtoolkit was added in #160040

**Pull request closed without merging**